src/test/java/com/datadog/api/ClientSteps.java

```diff
@@ -0,0 +1,279 @@
+        assertEquals(lookup(world.context, fixturePath), lookup(responseData, responsePath));
+    }
+
+    @Then("the response {string} has length {long}")
+    public void theResponseHasLength(String responsePath, Long size) throws java.lang.reflect.InvocationTargetException,
+            java.lang.IllegalAccessException, java.lang.InstantiationException, java.lang.NoSuchMethodException,
+            java.lang.ClassNotFoundException, java.lang.NoSuchFieldException {
+        Object responseData = world.responseClass.getMethod("getData").invoke(world.response);
+        List value = (List) lookup(responseData, responsePath);
+        assertEquals(size, Long.valueOf(value.size()));
+    }
+
+    /*
+     * Convert an identifier to class name.
+     */
+    public static String toClassName(String identifier) {
+        return replace(identifier, Pattern.compile("([A-Z])([A-Z]+)([A-Z][a-z])"), m -> {
+            return m.group(1) + m.group(2).toLowerCase() + m.group(3);
+        });
+    }
+
+    /*
+     * Convert an identifier to method name.
+     */
+    public static String toMethodName(String identifier) {
+        return replace(identifier, Pattern.compile("^([A-Z])"), m -> {
+            return m.group(1).toLowerCase();
+        });
+    }
+
+    /*
+     * Convert an identifier to property name.
+     */
+    public static String toPropertyName(String identifier) {
+        identifier = replace(identifier, Pattern.compile("_(.)"), m -> {
+            return m.group(1).toUpperCase();
+        });
+        return replace(identifier, Pattern.compile("\\[(.)([^]]*)\\]"), m -> {
+            return m.group(1).toUpperCase() + m.group(2);
+        });
+    }
+
+    public String getTracingEndpoint() {
+        return "features";
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> T getPropertyValue(Object obj, String field)
+            throws java.lang.IllegalAccessException, java.lang.NoSuchFieldException {
+        Class<?> clazz = obj.getClass();
+        Field f = clazz.getDeclaredField(field);
+        f.setAccessible(true);
+        Object ret = f.get(obj);
+        return (T) ret;
+    }
+
+    /*
+     * Lookup value from generic data object based on given path.
+     *
+     * Path contains attributes names or map keys separated by dots or list indices
+     * in square brackets.
+     *
+     * Example: "foo.bar[1].baz" can be read as data.getFoo().getBar()[1]["baz"]
+     */
+    public static Object lookup(Object data, String path)
+            throws java.lang.IllegalAccessException, java.lang.NoSuchFieldException {
+        Object result = data;
+        for (String dotPart : Arrays.asList(path.split("(?<=\\.)"))) {
+            dotPart = dotPart.replaceAll("\\.", "");
+            for (String part : Arrays.asList(dotPart.split("(?<=\\[)"))) {
+                if (part.indexOf("]") != -1) {
+                    Integer index = Integer.parseInt(part.replaceAll("]", ""));
```

Comment:
you'll also need to remove `[` since it could be there if you try to access a two-dimensional array for instance: `data[0][0]`.
Or split directly on the `[` character instead of using the lookbehind regex